src/connection/server_support.c

```diff
@@ -1309,16 +1309,21 @@ css_start_shutdown_server ()
  *       css_initialize_server_interfaces before calling this function.
  */
 int
-css_init (THREAD_ENTRY * thread_p, char *message_to_master, int message_to_master_length, int port_id)
+css_init (THREAD_ENTRY * thread_p, const char *server_name, int port_id)
 {
   CSS_CONN_ENTRY *conn;
   int status = NO_ERROR;
+  int message_to_master_length = 0;
+  char *message_to_master = NULL;
 
-  if (message_to_master == NULL || port_id <= 0)
+  if (server_name == NULL || port_id <= 0)
     {
       return ER_FAILED;
     }
 
+  message_to_master = css_pack_message_to_master (server_name, &message_to_master_length);
+  assert (message_to_master != NULL);
```

Comment:
log entry and/or `return ER_FAILED` besides assert?
if NULL, it will fallthrough to `shutdown`(being that `css_connect_to_master_server` will also return NULL), but there should be a log entry at least explaining why the server went down

Reply:
changed `message_to_master` type to std::string; the null case was not really necessary. packing the message should never fail and `server_name` should never be null.